contracts/other/notional/Join1155.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import "@yield-protocol/vault-interfaces/IJoin.sol";
+import "@yield-protocol/utils-v2/contracts/token/IERC20.sol";
+import "@yield-protocol/utils-v2/contracts/token/MinimalTransferHelper.sol";
+import "@yield-protocol/utils-v2/contracts/access/AccessControl.sol";
+import "@yield-protocol/utils-v2/contracts/math/WMul.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256U128.sol";
+import "./ERC1155.sol"; // TODO: Move to yield-utils-v2
```

Comment:
do we want to do this before we release or nah?

Reply:
I'm a bit conflicted. Do we really want to copy ERC1155 from solmate to yield-utils-v2? Or we import solmate? Do we then deprecate yield-utils-v2 ERC20.sol and use solmate's instead?

Reply:
Implementing Solmate ERC20 into the `yield-utils-v2` repo sounds like something to consider but outside of scope for this pr.  As for the 1155 contract, I'm cool with doing it this way (since this is kind of a one-off contract anyways) and just removing the comment here. 

We could create a new ticket in the `yield-utils-v2` repo "Use Solmate" and next time we have to interact with a ERC20 or ERC1155 contract, we can think about the best way to do that?

Lemme know if you want me to make that ticket ^^

Reply:
done https://github.com/yieldprotocol/yield-utils-v2/issues/33